### PR TITLE
Added idField to the resource during the update operation.

### PR DESCRIPTION
**Pull request by andriikonstantinov**
The id field wasn't passed to the extension during update events.
Added new test verifying pre_update and pre_update_in_transaction id changes.

### CODE REVIEW

**Comment by Zimnx** [comment]
please gofmt your changes

File: server/resources/resource_management.go [hunk 1]
```diff
@@ -611,16 +611,21 @@ func UpdateResource(
...
 	if err := extension.HandleEvent(context, environment, "pre_update", resourceSchema.ID); err != nil {
 		return err
 	}
 
 	if resourceData, ok := context["resource"].(map[string]interface{}); ok {
+		delete(resourceData, "id")
 		dataMap = resourceData
 	}
-
+	log.Critical("I am here")
```
Comment (p-kozlowski) [inline server/resources/resource_management.go]: remove

File: server/resources/resource_management_test.go [hunk 2]
```diff
@@ -1163,6 +1167,29 @@ var _ = Describe("Resource manager", func() {
 			fakeIdentity = &middleware.FakeIdentity{}
 		})
 
+		Describe("Whether id is not empty during update and update_in_transaction", func() {
+			BeforeEach(func() {
+				javascriptCode := `if (context.resource.id == undefined || context.resource.id == ""){
```
Comment (p-kozlowski) [inline server/resources/resource_management_test.go]: use ===

File: server/resources/tmp.txt [hunk 3]
```diff
@@ -0,0 +1,168 @@
+Running Suite: Resources Suite
```
Comment (p-kozlowski) [inline server/resources/tmp.txt]: this file should not be here

File: server/resources/resource_management.go [hunk 4]
```diff
@@ -611,16 +611,21 @@ func UpdateResource(
 	if err != nil {
 		return ResourceError{err, err.Error(), Unauthorized}
 	}
-	context["resource"] = dataMap
-
+	var dataMapCopy = make(map[string]interface{})
```
Comment (p-kozlowski) [inline server/resources/resource_management.go]: why do you need a copy?

File: server/resources/resource_management_test.go [hunk 5]
```diff
@@ -1163,6 +1167,29 @@ var _ = Describe("Resource manager", func() {
...
+					throw new CustomException();
+				}`
+
+				events["pre_update"] = javascriptCode
+				events["pre_update_in_transaction"] = javascriptCode
+			})
+			It("Should receive id and tenat_id but should not update them", func() {
```
Comment (p-kozlowski) [inline server/resources/resource_management_test.go]: typo: tenant_id

**Pull request closed without merging**